Model/Sync/Category/Processor/ProcessByCategory.php

```diff
@@ -190,6 +191,10 @@ public function processEntities($retryCategoryIds = [], $storeId = null)
                 || $this->isSyncedCategoryMissingYotpoId($yotpoSyncedCategories, $magentoCategory)
             ) {
                 $response = $this->syncAsNewCollection($magentoCategory);
+                $isCategorySyncedSuccessfully = $this->config->isResponseIndicatesSuccess($response);
+                if ($isCategorySyncedSuccessfully) {
+                    $this->updateCategoryProductsForCollectionsProductsSync($categoryId);
+                }
```

Comment:
As discussed offline, the logic here is incorrect.
In this case, there will be no entries in the sync table, so there won't be anything to update.

Reply:
Changed it to to use catalog_category_product table instead.